tests/samplers/test_common.py

```diff
@@ -626,6 +627,76 @@ def test_wildcard_nested_in_wildcard(
+        cmd = parse("wearing __colors-${temp}__ suede shoes")
+        resolved_value =  str(next(sampling_context.generator_from_command(cmd)))
+        assert resolved_value == "wearing {wildcard_wrap}colors-temp{wildcard_wrap} suede shoes".format(wildcard_wrap=sampling_context.wildcard_manager.wildcard_wrap)
+
+    @pytest.mark.parametrize("sampling_context", sampling_context_lazy_fixtures)
+    def test_wildcard_with_multiple_variables(self, sampling_context: SamplingContext):
+        cmd = parse("${genus=mammals}${species=feline}__animals/${genus:reptiles}/${species:snakes}__")
+        resolved_value =  str(next(sampling_context.generator_from_command(cmd)))
+        if isinstance(sampling_context.default_sampler, RandomSampler):
+            assert  resolved_value in ["cat", "tiger"]
+        else:
+            assert resolved_value == "cat"
+
+    @pytest.mark.parametrize("sampling_context", sampling_context_lazy_fixtures)
+    def test_wildcard_with_variable_and_glob(self, sampling_context: SamplingContext):
+        cmd = parse("${genus=reptiles}__animals/${genus}/*__")
+        resolved_value =  str(next(sampling_context.generator_from_command(cmd)))
+        if isinstance(sampling_context.default_sampler, RandomSampler):
+            assert  resolved_value in ["cobra", "gecko", "iguana", "python"]
+        else:
+            assert resolved_value == "cobra"        
+    
+    @pytest.mark.parametrize("sampling_context", sampling_context_lazy_fixtures)
+    def test_wildcard_with_variable_in_nested_wildcard(self, sampling_context: SamplingContext):
+        cmd = parse("${genus=mammals}__animal__")
+        resolved_value =  str(next(sampling_context.generator_from_command(cmd)))
+        if isinstance(sampling_context.default_sampler, RandomSampler):
+            assert  resolved_value in ["cat", "tiger", "dog", "wolf"]
+        else:
+            assert resolved_value == "cat"
+
+    @pytest.mark.parametrize("sampling_context", sampling_context_lazy_fixtures)
+    def test_nested_wildcard_with_parameterized_variable(self, sampling_context: SamplingContext):
+        cmd = parse("__animal(genus=mammals)__")
+        resolved_value =  str(next(sampling_context.generator_from_command(cmd)))
+        if isinstance(sampling_context.default_sampler, RandomSampler):
+            assert  resolved_value in ["cat", "tiger", "dog", "wolf"]
+        else:
+            assert resolved_value == "cat"        
+        
+    @pytest.mark.parametrize("sampling_context", sampling_context_lazy_fixtures)
+    def test_nested_wildcards_in_single_file(self, sampling_context: SamplingContext):
+        cmd = parse("${car=!{porsche|john_deere}}a __cars/${car}/types__ made by __cars/${car}/name__")
+        resolved_value =  str(next(sampling_context.generator_from_command(cmd)))
+        assert  resolved_value in [
+            "a sports car made by Porsche",
+            "a tractor made by John Deere"
+        ]
```

Comment:
Using a tuple is slightly better here since Python can then do clever bytecode things (when it can be assured the list can't be mutated).
```suggestion
        assert resolved_value in (
            "a sports car made by Porsche",
            "a tractor made by John Deere"
        )
```